### ISSUE TITLE
RC (Release Candidate) Generation process by Decoupling Release Notes

### ISSUE THREAD

**Issue by prudhvigodithi** [opensearch-project/opensearch-build#5065]
### Is your feature request related to a problem? Please describe

For today's release, having component release notes is an entry criteria, meaning the release notes must be merged into the release branch before generating the release candidate (RC). In past releases, this requirement was not strictly followed, as missing release notes could be a blocker for RC generation, preventing teams to test the RC. As a result, RC builds have been started even when some components were missing the release notes. This adds extra work for the release manager, who must follow up on missing release notes, and puts in situation where the locked RC commit ends up missing these notes. 

Automations, such as the [release notes check workflow](https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/release-workflows/release-notes-check.jenkinsfile), may fail because the input manifest includes a component commit that lacks release notes, leading to manual interventions like [this PR which has the consolidated release notes](https://github.com/opensearch-project/opensearch-build/pull/5063).

Finalizing the RC before all entry criteria are met is not a good practice, but blocking the RC build until all plugin teams provide release notes is also inefficient, as it delays plugins that are ready for release. 

### Describe the solution you'd like

- Plugins should develop the habit of following to the entry criteria and adding the release notes. 

- Maintain a new branch called `release-notes` where all plugins must add their release notes. This commit will not, and need not, be part of the RC, as it is neither a feature nor a bug fix. Moving forward, automations can use the `release-notes` branch to check version-specific release notes. The RC can be generated and finalized only after meeting the entry criteria of having the release notes, and having a separate `release-notes` branch ensures that the RC doesn't need to be rebuilt, since the branches are now independent.

- For example, if we start the build and later declare it as the RC, which is often done for patch releases, the RC need not be rebuilt just because the release notes were initially missing (adding release notes creates a new commit). With the release notes now added to the `release-notes` branch, the RC can be finalized, as the entry criteria of having release notes is met, without introducing a new commit to the release branch, avoiding unnecessary rebuilds.

**Comment by prudhvigodithi** [comment]
Adding @dblock @getsaurabh02 @peterzhuamazon @gaiksaya @rishabh6788 to provide some inputs.

**Comment by peterzhuamazon** [comment]
> Adding @dblock @getsaurabh02 @peterzhuamazon @gaiksaya @rishabh6788 to provide some inputs.

The issue I see is if user use the bundle manifest to get the commit, those commit will not contain the release notes.
I dont think we need to create a new branch to do that which will add more confusion.

**Comment by peterzhuamazon** [comment]
> > Adding @dblock @getsaurabh02 @peterzhuamazon @gaiksaya @rishabh6788 to provide some inputs.
> 
> The issue I see is if user use the bundle manifest to get the commit, those commit will not contain the release notes. I dont think we need to create a new branch to do that which will add more confusion.

Thanks.

**Comment by gaiksaya** [comment]
Consider using `major.minor` branch for consolidating the release notes rather than a fix commit used in manifest file. 
Open to more suggestions.

**Comment by gaiksaya** [comment]
Closing this issue as we aim to still keep the release notes addition as an entrance criteria and check regularly to get that in.
